File: app/src/main/java/com/secondmind/minimal/MainActivity.java
package com.secondmind.minimal;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    TextView tv = new TextView(this);
    tv.setText("SecondMind Minimal Base");
    tv.setTextSize(20f);
    setContentView(tv);
  }
}

File: app/src/core/java/com/secondmind/minimal/MainActivity.kt
package com.secondmind.minimal

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

class MainActivity : ComponentActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContent {
      MaterialTheme {
        Surface(modifier = Modifier.fillMaxSize()) {
          Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Column(horizontalAlignment = Alignment.CenterHorizontally,
                   verticalArrangement = Arrangement.spacedBy(12.dp)) {
              Text("SecondMind Core", style = MaterialTheme.typography.titleLarge)
              var clicks by remember { mutableStateOf(0) }
              Button(onClick = { clicks++ }) { Text("Tap ($clicks)") }
              Text("This is the minimal core build (Compose only).")
            }
          }
        }
      }
    }
  }
}
